### ISSUE TITLE
Prod: Query create issue for composition queries when selecting DV_QUANTITY paths

### ISSUE THREAD

**Issue by ppazos** [ppazos/cabolabs-ehrserver#553]
server001.cloudehrserver.com 
Create a query
Select blood pressure
Select any DV_QUANTITY path

It is not possible to set a range/between condition for the query!

The console shows
Cannot get property 'list' on null object. Stacktrace follows:
java.lang.NullPointerException: Cannot get property 'list' on null object
	at com.cabolabs.ehrserver.query.datatypes.DataCriteriaDV_QUANTITY.criteriaSpec(DataCriteriaDV_QUANTITY.groovy:70)
	at com.cabolabs.ehrserver.query.QueryController.getCriteriaSpec(QueryController.groovy:452)
	at grails.plugin.cache.web.filter.PageFragmentCachingFilter.doFilter(PageFragmentCachingFilter.java:198)
	at grails.plugin.cache.web.filter.AbstractFilter.doFilter(AbstractFilter.java:63)
	at grails.plugin.springsecurity.web.filter.GrailsAnonymousAuthenticationFilter.doFilter(GrailsAnonymousAuthenticationFilter.java:53)
	at grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.doFilter(RequestHolderAuthenticationFilter.java:53)
	at grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.java:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

**Comment by ppazos** [comment]
On dev this is working OK.

The problem might be related to OptManager, since DV_QUANTITY is the only one that uses it for criteriaSpec(), and I did a hot deploy on Tomcat (copy the WAR without restarting the service).

Since OptManager is Singleton, it might not get reloaded when the bootstrap run and it should.

**Comment by ppazos** [comment]
The real problem was I deleted the opts to update them, but since there are records of those OPT on the database, the new ones were not updated.

With no OPT in /opts. the OptManager didn't loaded anything.

But still, I think OptManager maintained old data in the cache, so previous comment is still valid but wasn't the main issue. I did a Generate Indexes, and the database records were erased since no OPTs are in /opt.

Now trying to redeploy to see if the OPTs get loaded and the OptManager gets updated.
